lib/db.c

```diff
@@ -13028,8 +13028,13 @@ grn_ctx_at(grn_ctx *ctx, grn_id id)
             grn_ja_unref(ctx, &iw);
           }
           if (grn_enable_reference_count) {
-            if (!vp->ptr) {
+            if (vp->ptr->header.type == GRN_TYPE
+                || vp->ptr->header.type == GRN_EXPR) {
```

Comment:
We should add a comment why we need to have special case for them.

BTW, I think that we don't need `GRN_EXPR` here because we don't use permanent `GRN_EXPR` now. Our `GRN_EXPR` is always temporal now.